### PR TITLE
Add documentation and guide for tracing schema

### PR DESCRIPTION
**Pull request by BenWilson2**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/BenWilson2/mlflow/pull/12521?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12521/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12521
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Adds docs for tracing schema and an explanation of trace architecture

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [ ] Examples
  - [X] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: docs/source/_static/images/llms/tracing/schema/span_architecture.png
Comment (harupy) [inline docs/source/_static/images/llms/tracing/schema/span_architecture.png:1]: Can we just use a table or list?

File: docs/source/_static/images/llms/tracing/schema/trace_info_architecture.png
[the remark this answers is not in the document]
Comment (B-Step62) [reply docs/source/_static/images/llms/tracing/schema/trace_info_architecture.png:1]: +1, I like those images, but given that we are still pretty much in development phase, easiness of update is important.

File: docs/source/llms/tracing/tracing-schema.rst [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+To learn more about how ``TraceInfo`` metadata is used for searching, you can see examples :ref:`here <search_traces>`.
+
+The data that is contained in the ``TraceInfo`` object is used to populate the trace view page within the MLflow tracking UI, as shown below.
+
+.. figure:: ../../_static/images/llms/tracing/schema/trace_info_in_ui.png
+    :alt: TraceInfo as it is used in the MLflow UI
+    :width: 100%
+    :align: center
+
+The primary components of MLflow :py:class:`mlflow.entities.trace_info.TraceInfo` objects are listed below.
```
Comment (B-Step62) [inline docs/source/llms/tracing/tracing-schema.rst:125]:
```suggestion
The primary components of MLflow :py:class:`TraceInfo <mlflow.entities.trace_info.TraceInfo>` objects are listed below.
```

File: docs/source/llms/tracing/tracing-schema.rst [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+Request ID
+^^^^^^^^^^
+
+The ``request_id`` of a trace is a unique identifier that is generated for each trace. This identifier is used within MLflow and integrated systems to 
+resolve the event being captured and to provide associations for external systems to map the logged trace to the originating caller. 
+
+.. note::
+    In the fluent API, this value will be generated for you. In the client API, you can provide a request_id to associate with the trace or omit
```
Comment (B-Step62) [inline docs/source/llms/tracing/tracing-schema.rst:134]: This is a bit inaccurate. Regardless of which API they use, the request ID must be generated by MLflow (`MlflowClieit().start_trace()` generates request ID). What users can control is which generated ID they can pass to each spans.

File: docs/source/llms/tracing/tracing-schema.rst [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+When searching through traces, this field is critical for filtering and grouping traces by experiment.
+
+
+Timestamp (ms)
+^^^^^^^^^^^^^^
+
+The ``timestamp_ms`` property within TraceInfo marks the time at which the trace was created. This is a Unix timestamp in milliseconds.
+
+.. note::
+    The time reflected here is the time at with the trace was created, not the time at which a request to your application was made. As such,
```
Comment (B-Step62) [inline docs/source/llms/tracing/tracing-schema.rst:151]: +1 to have this noted in the doc!

File: docs/source/llms/tracing/tracing-schema.rst [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+The request metadata are additional key-value pairs of information that are associated with the Trace, set and modified by the tracing backend. 
+These are not open for addition or modification by the user, but can provide additional context about the trace, such as an MLflow ``run_id`` that is 
+associated with the trace. 
+
+This metadata is immutable and considered system-controlled.
+
+Tags
+^^^^
+
+User-defined metadata that can be applied to a trace for applying additional context, aid in :ref:`search functionality <search_traces>`, or to
```
Comment (B-Step62) [inline docs/source/llms/tracing/tracing-schema.rst:185]:
```suggestion
User-defined key-value pairs that can be applied to a trace for applying additional context, aid in :ref:`search functionality <search_traces>`, or to 
```

nit: cosmetic, but maybe we can avoid the term "metadata"

File: docs/source/llms/tracing/tracing-schema.rst [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+        .. tab:: Trace Data
+
+            |trace-data|
+
+        .. tab:: Span Architecture
+
+            |span-architecture|
+
+
+Trace Schema
```
Comment (B-Step62) [inline docs/source/llms/tracing/tracing-schema.rst:96]: Will it be messy if we convert these sections as tables, probably one for each high-level object (TraceInfo / TraceData / Span)? The rational is that the h2 and h3 is hardly distinguishable in Sphinx, so it is not super easy to grasp the data structure from flat text sections. WDYT?

File: docs/source/llms/tracing/tracing-schema.rst [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+SpanEvent
+~~~~~~~~~
+
+The :py:class:`Span Event <mlflow.entities.SpanEvent>` object, if present, consists of three elements:
+
+- **name** - defines the name of the event that occurred.
+- **timestamp** - the time at which the event occurred in microseconds.
+- **attributes** - a key-value collection of information related to the event that occurred. In the event of an exception being thrown, the stack trace 
+  will be included in this collection to aid in debugging and diagnosing the root cause of issues.
```
Comment (B-Step62) [inline docs/source/llms/tracing/tracing-schema.rst:276]:
```suggestion
```

nit: this seems to be repeating https://mlflow.org/docs/latest/python_api/mlflow.entities.html#mlflow.entities.SpanEvent (which is already linked)

File: docs/source/_static/images/llms/tracing/schema/trace_info_architecture.png
[the remark this answers is not in the document]
Comment (xq-yin) [reply docs/source/_static/images/llms/tracing/schema/trace_info_architecture.png:1]: idk, I find visualization extremely powerful and would like to keep it. we could add note there to say the diagram might not be up to date and add a text table below (so easier to update).

File: docs/source/llms/tracing/tracing-schema.rst [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+
+.. note::
+    The time reflected here is the time at with the trace was created, not the time at which a request to your application was made. As such, 
+    it does not factor into account the time it took to process the request to the environment in which your application is being served, which 
+    may introduce additional latency to the total round trip time, depending on network configurations. 
+
+Execution Time (ms)
+^^^^^^^^^^^^^^^^^^^
+
+The ``execution_time_ms`` property within TraceInfo marks the total time that the trace took to complete. This is a Unix timestamp in milliseconds.
```
Comment (harupy) [inline docs/source/llms/tracing/tracing-schema.rst:158]:
```suggestion
The ``execution_time_ms`` property within ``TraceInfo`` marks the total time that the trace took to complete. This is a Unix timestamp in milliseconds.
```

File: docs/source/llms/tracing/tracing-schema.rst (hunk 2 above)
Comment (BenWilson2) [reply docs/source/llms/tracing/tracing-schema.rst:134]: ahhh yeah. Fixed this

File: docs/source/llms/tracing/tracing-schema.rst (hunk 5 above)
Comment (BenWilson2) [reply docs/source/llms/tracing/tracing-schema.rst:96]: Makes sense. It's a bit more dense, but I'm going with the table approach - it's a bit more concise

File: docs/source/llms/tracing/tracing-schema.rst (hunk 6 above)
Comment (BenWilson2) [reply docs/source/llms/tracing/tracing-schema.rst:276]: refactoring this

File: docs/source/llms/tracing/tracing-schema.rst (hunk 7 above)
Comment (BenWilson2) [reply docs/source/llms/tracing/tracing-schema.rst:158]: reworded to remove the reference

File: docs/source/_static/images/llms/tracing/schema/trace_info_architecture.png
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/source/_static/images/llms/tracing/schema/trace_info_architecture.png:1]: We could, but we've been getting a lot of positive feedback on these images that help to explain dense text. It seems like people like them for a 'quick check' of a concept without having to read so much text

[the remark this answers is not in the document]
Comment (harupy) [reply docs/source/_static/images/llms/tracing/schema/trace_info_architecture.png:1]: For this particular image, I don't see a need to use an image.

**Review by harupy — APPROVED** [review]
Stamp to unblock 2.14.2 release. We might want to introduce https://mermaid.js.org/intro/ to avoid using too many images.

**Pull request merged**